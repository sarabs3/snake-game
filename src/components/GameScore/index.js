import React from 'react';

const GameScore = props => {
	return <div className={'gameScore'} data-testid={'game-score'}>Score: <b>{props.score}</b></div>
};

export default GameScore;
