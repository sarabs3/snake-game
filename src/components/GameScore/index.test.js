import React from 'react';
import ReactDOM from 'react-dom';

import {render} from '@testing-library/react';
import renderer from 'react-test-renderer'

import GameScore from './index'

let div;

beforeEach(() => {
	div = document.createElement('div');
	document.body.appendChild(div);
});

afterEach(() => {
	ReactDOM.unmountComponentAtNode(div);
	div.remove();
	div = null;
});

it('<GameScore /> renders without crashing', () => {
	const {queryByTestId} = render(<GameScore/>, div);

	expect(queryByTestId('game-score')).toBeTruthy();
});

it('<GameScore /> renders correctly', () => {
	const {queryByTestId} = render(<GameScore score={10}/>, div);

	expect(queryByTestId('game-score').textContent).toBe('Score: 10');
});

it('<GameScore /> snapshot matched', () => {
	const tree = renderer.create(<GameScore score={10} />, div).toJSON();

	expect(tree).toMatchSnapshot();
});
