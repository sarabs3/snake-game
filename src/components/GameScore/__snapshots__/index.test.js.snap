// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameScore /> snapshot matched 1`] = `
<div
  className="gameScore"
  data-testid="game-score"
>
  Score: 
  <b>
    10
  </b>
</div>
`;
